### PR TITLE
Added REST API endpoints

### PR DESCRIPTION
**Pull request by innomaxx**
Default path : http://localhost:8080

+ /menu -> get cafe menu
+ /orders/new -> make an order, receive receipt

Also added separate models for API. It's redundant ofc, will search how to serialize models, generated from proto files, to JSON.

### CODE REVIEW

File: src/main/java/com/max/grpc/orders/client/rest/RestServer.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        if (server != null || server.isStarted()) {
+            server.shutdownNow();
+            logger.info("Server stopped");
+        } else {
+            logger.info("Server is already stopped or hasn't started yet");
+        }
+    }
+
+    private URI getBaseURI() {
+        return UriBuilder.fromUri("http://localhost/").port(port).build();
```
Comment (GalynaY) [inline src/main/java/com/max/grpc/orders/client/rest/RestServer.java:48]: Port should be configurable as well not hard coded

File: src/main/java/com/max/grpc/orders/client/rest/models/ApiFoodItem.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    public int getPrice() {
+        return price;
+    }
+
+    public void setPrice(int price) {
+        this.price = price;
+    }
+
+    public static ApiFoodItem from(FoodItem protoItem) {
+        var responseItem = new ApiFoodItem();
```
Comment (GalynaY) [inline src/main/java/com/max/grpc/orders/client/rest/models/ApiFoodItem.java]: Good! If you want less code can try to use  MapStract. Great lib for this

File: src/main/java/com/max/grpc/orders/client/rest/services/OrderService.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import java.util.List;
+
+@Path("/orders")
+public interface OrderService {
+    @POST
+    @Path("/new")
+    @Consumes(MediaType.APPLICATION_JSON)
```
Comment (GalynaY) [inline src/main/java/com/max/grpc/orders/client/rest/services/OrderService.java]: Why do you need interfaces?

File: src/main/java/com/max/grpc/orders/client/rest/services/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    @Path("/new")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Override
+    public ApiOrderReceipt makeOrder(List<String> itemIds) {
+        logger.info("Making order, " + itemIds.size() + " items");
+        OrderReceipt receipt = cafeClient.makeOrder(itemIds);
+        logger.info("Order made, receipt received");
+
+        ApiOrderReceipt responseReceipt = new ApiOrderReceipt();
```
Comment (GalynaY) [inline src/main/java/com/max/grpc/orders/client/rest/services/OrderServiceImpl.java]: This code is mapper logic not for endpoint. Move it to mapper

File: src/main/java/com/max/grpc/orders/client/rest/services/MenuServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    @GET
+    @Path("/")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Override
+    public ApiCafeMenu getMenu() {
+        logger.info("Getting menu from cafe...");
+        CafeMenu cafeMenu = cafeClient.getMenu();
+        logger.info("Received " + cafeMenu.getItemsCount() + " dishes");
+
+        ApiCafeMenu responseMenu = new ApiCafeMenu();
```
Comment (GalynaY) [inline src/main/java/com/max/grpc/orders/client/rest/services/MenuServiceImpl.java]: Logic for the mapper

File: src/main/java/com/max/grpc/orders/client/rest/services/MenuServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Path("/menu")
+public class MenuServiceImpl implements MenuService {
```
Comment (GalynaY) [inline src/main/java/com/max/grpc/orders/client/rest/services/MenuServiceImpl.java]: This is Endpoint, Api ... whatever but Service. You need a Service or at least a mapper to do the job. The endpoint should receive that Service and operate with it. It should not know about gRPC, proto or any other details of the implementation

**Review by GalynaY — CHANGES_REQUESTED** [review]

File: src/main/java/com/max/grpc/orders/client/rest/services/OrderService.java (hunk 3 above)
Comment (innomaxx) [reply src/main/java/com/max/grpc/orders/client/rest/services/OrderService.java]: Copied from one of examples, I thought it's a good practice ¯ \ _ (ツ) _ / ¯
I'll remove it

File: src/main/java/com/max/grpc/orders/client/rest/RestServer.java (hunk 1 above)
Comment (innomaxx) [reply src/main/java/com/max/grpc/orders/client/rest/RestServer.java:48]: Done

File: src/main/java/com/max/grpc/orders/client/rest/services/OrderService.java (hunk 3 above)
Comment (innomaxx) [reply src/main/java/com/max/grpc/orders/client/rest/services/OrderService.java]: Removed

File: src/main/java/com/max/grpc/orders/client/rest/models/ApiFoodItem.java (hunk 2 above)
Comment (innomaxx) [reply src/main/java/com/max/grpc/orders/client/rest/models/ApiFoodItem.java]: Switched

File: src/main/java/com/max/grpc/orders/client/rest/services/MenuServiceImpl.java (hunk 5 above)
Comment (innomaxx) [reply src/main/java/com/max/grpc/orders/client/rest/services/MenuServiceImpl.java]: Done

File: src/main/java/com/max/grpc/orders/client/rest/services/OrderServiceImpl.java (hunk 4 above)
Comment (innomaxx) [reply src/main/java/com/max/grpc/orders/client/rest/services/OrderServiceImpl.java]: Done

File: src/main/java/com/max/grpc/orders/client/rest/services/MenuServiceImpl.java (hunk 6 above)
Comment (innomaxx) [reply src/main/java/com/max/grpc/orders/client/rest/services/MenuServiceImpl.java]: Done

**Pull request merged**